eslint.config.mjs

```diff
@@ -0,0 +1,145 @@
+import eslint from "@eslint/js";
+import tseslint from "@typescript-eslint/eslint-plugin";
+import tsParser from "@typescript-eslint/parser";
+import prettierConfig from "eslint-config-prettier";
```

Comment:
we don't need to explicitly add prettierConfig as eslint-plugin-prettier sets them both
https://github.com/prettier/eslint-plugin-prettier?tab=readme-ov-file#configuration-new-eslintconfigjs